pallets/polkadex-ido/src/lib.rs

```diff
@@ -664,19 +665,18 @@ pub mod pallet {
             Ok(())
         }
 
-        /// Stores information about investors, showing interest in funding round.
+        /// Allows investors to invest in an active IDO 
         ///
         /// # Parameters
         ///
         /// * `round_id`: Funding round id
+        /// * `amount`: Amount to invest in TokenB 
         #[pallet::weight((10_000, DispatchClass::Normal))]
-        pub fn show_interest_in_round(origin: OriginFor<T>, round_id: T::Hash, amount: BalanceOf<T>) -> DispatchResult {
+        pub fn invest(origin: OriginFor<T>, round_id: T::Hash, amount: BalanceOf<T>) -> DispatchResult {
             let investor_address: T::AccountId = ensure_signed(origin)?;
             ensure!(<InfoInvestor<T>>::contains_key(&investor_address), <Error<T>>::InvestorDoesNotExist);
-            ensure!(!<InfoFundingRound<T>>::contains_key(&round_id), Error::<T>::FundingRoundNotApproved);
```

Comment:
I would suggest to leave this assertion since this is basically application of quick return which is good, besides `contains_key` potentially chipper query than aka `SELECT` from 679 line.